beacon_node/network/src/beacon_processor/worker/gossip_methods.rs

```diff
@@ -1015,6 +1025,10 @@ impl<T: BeaconChainTypes> Worker<T> {
                 );
             }
         };
+
+        if result.is_err() {
```

Comment:
I wonder if storing the `BlockError` as well would help with debugging?

Reply:
IMO we don't need to store the block error because it's already logged above. Although it could be convenient.

My concern is that this PR currently counts EL timeouts as invalid blocks and will store the block SSZ unnecessarily. I guess this is OK if we don't mind the invalid block storage containing a few false positives, although it would hamper an effort to analyze _every_ block in the directory. I suspect in practice we'll use the invalid block storage to look up specific blocks based on `Invalid gossip beacon block` log messages or manual reports. In this case, it doesn't matter if there's some junk in the storage, apart from it being a waste of space.

Maybe we could just filter out the most common false positive (`TimedOut`) and allow the rest? I think this would be quite pragmatic using a function like `is_timeout` on `ExecutionPayloadError`

Reply:
Oh I figured out how to reply to this comment. The same comment appears a second time, but without the reply field 🤷 

Perhaps writing out the error message is a nice happy medium about the timeout stuff mentioned in https://github.com/sigp/lighthouse/pull/4194#issuecomment-1547209896. Super easy access to the error message might make those timeouts easier to ignore. I'll add it in.